temporal/api/cloud/cloudservice/v1/request_response.proto

```diff
@@ -234,3 +235,89 @@ message GetRegionResponse {
     // The temporal cloud region.
     temporal.api.cloud.region.v1.Region region = 1;
 }
+
+message GetNexusEndpointsRequest {
+    // The requested size of the page to retrieve - optional.
+    // Cannot exceed 1000. Defaults to 100. 
+    int32 page_size = 1;
+
+    // The page token if this is continuing from another response - optional.
+    string page_token = 2;
+
+    // Filter endpoints based on their target specification - optional, treated as an AND if specified. Empty fields within the spec are treated as '*'.
+    temporal.api.cloud.nexus.v1.EndpointTargetSpec spec = 3;
+
+    // Filter endpoints by their name - optional, treated as an AND if specified. Specifying this will result in zero or one results.
+    string name = 4;
+}
+
+message GetNexusEndpointsResponse {
+    // The list of endpoints in ascending id order.
+    repeated temporal.api.cloud.nexus.v1.Endpoint endpoints = 1;
+
+    // The next page's token.
+    string next_page_token = 2;
+}
+
+message GetNexusEndpointRequest {
+    // The id of the nexus endpoint to get.
+    string endpoint_id = 1;
+}
+
+message GetNexusEndpointResponse {
+    // The nexus endpoint.
+    temporal.api.cloud.nexus.v1.Endpoint endpoint = 1;
+}
+
+message CreateNexusEndpointRequest {
+    // The spec for the nexus endpoint.
+    temporal.api.cloud.nexus.v1.EndpointSpec spec = 1;
+
+    // The id to use for this async operation - optional.
```

Comment:
Do we have guidance anywhere explaining why you'd want to specify this? I understand the concept but I'm not sure users do.

Reply:
Agree that this probably needs to be fleshed out more here: https://docs.temporal.io/ops. The API itself is still in preview.  I have seen other APIs have a section about "idempotency" which describes the usage/semantics around similar fields. We may want something similar her.

cc @anekkanti @jlacefie in case there is some other documentation I am missing here, or to potentially track ensuring docs cover this eventually

Reply:
@rachfop since he is working on docs